osu.Game.Rulesets.Osu/Mods/OsuModAutopilot.cs

```diff
@@ -36,21 +39,63 @@ public class OsuModAutopilot : Mod, IApplicableFailOverride, IUpdatableByPlayfie
 
         private int currentFrame;
 
+        int cachedFrames = 1;
+        private const float relax_leniency = 3f;
+
         public void Update(Playfield playfield)
         {
             if (currentFrame == replayFrames.Count - 1) return;
 
-            double time = gameplayClock.CurrentTime;
+            double time = playfield.Clock.CurrentTime;
+
+            bool breakNow = false;
+            bool pause = false;
```

Comment:
what do these mean/do?

Reply:
breakNow is intended to break out of the for loop below. In the switch statement, when a slider or a hit circle is encountered it will break out of the loop so that it does not the other circles that come after it. 

pause is intended to pause the cursor (pausing the frames and storing the paused frames in the cachedFrames variable) for a few frames to allow leniency for the player. For instance if the player clicks a bit later than accurate it will still register abiet as a meh or good.